eclass/debian.eclass

```diff
@@ -0,0 +1,82 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: debian.eclass
+# @MAINTAINER:
+# Sam James <sam@gentoo.org>
+# @SUPPORTED_EAPIS: 8
+# @BLURB: Convenience eclass for easily using Debian patchsets as upstream
+# @DESCRIPTION:
+# TODO
+
+# @ECLASS_VARIABLE: DEBIAN_PN
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Debian name for the package. Defaults to ${PN}.
+: ${DEBIAN_PN:=${PN}}
+
+# @ECLASS_VARIABLE: DEBIAN_PV
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Debian patch level. Defaults to ${PV/*_p}.
+: ${DEBIAN_PV:=${PV/*_p}}
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls whether the "original" upstream sources (unpatched) are fetched
+# from Debian or not. Defaults to no.
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG_PV
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls the PV used without the patchlevel. Defaults to ${PV/_p*}.
+: ${DEBIAN_ORIG_PV:=${PV/_p*}}
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG_EXT
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls the Debian "original" upstream sources (unpatched) file extension.
+# Defaults to gz.
+: ${DEBIAN_ORIG_EXT:=.gz}
```

Comment:
You want:

```suggestion
: ${DEBIAN_ORIG_EXT:=gz}
```

I think?

It's referenced like `tar.${DEBIAN_ORIG_EXT}` later so the extension ends up double-dotted, and your use of this var in the ebuild later doesn't include the leading `.`.

Reply:
oh, good spot -- I've changed it to include . for now but we might flip it back (consistently this time)